hdhomerun_sock_posix.c

```diff
@@ -35,54 +33,39 @@
 #define MSG_NOSIGNAL 0
 #endif
 
+
 struct hdhomerun_sock_t {
 	int sock;
 };
 
 int hdhomerun_local_ip_info(struct hdhomerun_local_ip_info_t ip_info_list[], int max_count)
 {
-	int sock = socket(AF_INET, SOCK_DGRAM, 0);
+	int sock = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
 	if (sock == -1) {
 		return -1;
 	}
 
-	struct ifconf ifc;
-	size_t ifreq_buffer_size = 1024;
-
-	while (1) {
-		ifc.ifc_len = (int)ifreq_buffer_size;
-		ifc.ifc_buf = (char *)malloc(ifreq_buffer_size);
-		if (!ifc.ifc_buf) {
-			close(sock);
-			return -1;
-		}
-
-		memset(ifc.ifc_buf, 0, ifreq_buffer_size);
-
-		if (ioctl(sock, SIOCGIFCONF, &ifc) != 0) {
-			free(ifc.ifc_buf);
-			close(sock);
-			return -1;
-		}
-
-		if (ifc.ifc_len < (int)ifreq_buffer_size) {
-			break;
-		}
-
-		free(ifc.ifc_buf);
-		ifreq_buffer_size += 1024;
+	/*
+	 * Retrieve network interfaces from OS. if_nameindex() dynamically allocates a list of 
+	 * network interface names and indexs. 
+	 */
+	struct if_nameindex *if_ni, *iter;
+	if_ni = if_nameindex();
+	if (if_ni == NULL) {
+		perror("if_nameindex");
+		exit(EXIT_FAILURE);
```

Comment:
Calling `exit()` in a library is not good form. It prevents the error from being recovered from.

Reply:
```suggestion
		return -1;
```